game-core/src/main/java/games/strategy/debug/ErrorMessage.java

```diff
@@ -92,10 +93,10 @@ public static void enable() {
     INSTANCE.enableErrorPopup = true;
   }
 
-  public static void show(final String msg) {
+  public static void show(final LogRecord record) {
```

Comment:
Is this where you plan on initiating the report to the HTTP server?  If so, we may want to reconsider that.  This class is a UI component, and it probably shouldn't have responsibility for sending stuff over the network.

We now have three sinks for logs:

1. The console
1. The `ErrorMessage` UI
1. The HTTP error reporting service

Maybe we should take a step back and install multiple handlers into the `Logger` for each of these sinks instead of having everything piggyback off of `ConsoleHandler`?

Reply:
In one manner, but not quite as a stream. For a first and second rev having something that can be used as a generic error reporting tool independent of an error seems useful. We can attach some nice functionality to that like upload save game and screenshots, but fundamentally the upload is user driven and they can attach repro/description text to the error message.

In this location the intent was to add a button that would pass the `LogRecord` data to it's action handler if user chooses a 3rd option instead of closing an error or viewing details, they can upload.

Reply:
No worries, I think the original thinking was a behind the scenes error reporter. A stream of errors may not be that useful, attaching user-messages will help the quality of the error report and also allows us to have a generic in-game reporting feature. That was a pretty late-breaking shift in the intended feature.